### ISSUE TITLE
Port comma-delimited X-Forwarded-Proto support from deis/deis

### ISSUE THREAD

**Issue by krancour** [deis/router#155]
Port deis/deis#4998 to this project.

**Comment by krancour** [comment]
So far, the v2 router seems unaffected by issue deis/deis#4997.  I'd like to get a better handle on why before concluding that deis/deis#4998 does not need to be ported forward.

**Comment by krancour** [comment]
Ok... glad I took the time to investigate this.  I'm unable to reproduce deis/deis#4997 on v2 _using the same procedure and tests apps as were used for v1.x/LTS_.  In deis/deis#4997, the test "proxy-frontend" service gateway @gerred came up with would (via hapi) append to the end of the `X-Forwarded-Proto` header and when that request went out to the internet and came back in through the ELB to the router (to reach the backend service), `https,https` is the value that the router would see for that header and that was the proximate cause of the redirect loop described in deis/deis#4997.  This is _not_ happening as I expect in v2.  The router ends up seeing and `X-Forwarded-Proto` value of just `https`-- which is just fine and doesn't trigger the infinite redirect loop.

Initially, I was chalking this up to some aspect of k8s networking that I must not understand as well as I wish I did.  Perhaps, for instance, the kube-proxy played a role in canonicalizing the value of request headers?  Idk.

But... as it turns out, I _can_ reproduce the issue in v2 through other means.  If I `curl -H "X-Forwarded-Proto: https,https" https://backend-service.krancour.deis.ninja`, the router _does_ see an `X-Forwarded-Proto` header with value `https,https`, which != `https`-- which in turn triggers the undesired redirect.

To test the theory that it was the kube-proxy somehow canonicalizing the values of the request headers, I exec'ed into the container of a running app and used the same `curl` command as above.  Once again, I _am_ able to reproduce the problem.

At this point I'm left scratching my head as to why the steps used to demonstrate the problem on v1.x won't reproduce it on v2, but since the problem can be demonstrated through other means, I do believe we should carry on with porting deis/deis#4998 to v2.

fwiw... the method I have used to successfully reproduce the issue in v2 is likely to be a condition that can occur in the wild.  For instance, any time there is a proxy of some sort between the end user and the router's load balancer, depending on what exactly that proxy does, this issue could manifest.  I actually am highly suspicious that that is exactly what has occurred in deis/deis#4667

Proceeding with the change...
